code/modules/asset_cache/spritesheet/legacy/legacy_spritesheet.dm

```diff
@@ -0,0 +1,418 @@
+
+// spritesheet implementation - coalesces various icons into a single .png file
+// and uses CSS to select icons out of that file - saves on transferring some
+// 1400-odd individual PNG files
+#define SPR_SIZE 1
+#define SPR_IDX 2
+#define SPRSZ_COUNT 1
+#define SPRSZ_ICON 2
+#define SPRSZ_STRIPPED 3
+
+/// Deprecated: Use /datum/asset/spritesheet_batched where possible
+/datum/asset/spritesheet
+	_abstract = /datum/asset/spritesheet
+	cross_round_cachable = TRUE
+	var/name
+	/// List of arguments to pass into queuedInsert
+	/// Exists so we can queue icon insertion, mostly for stuff like preferences
+	var/list/to_generate = list()
+	var/list/sizes = list()    // "32x32" -> list(10, icon/normal, icon/stripped)
+	var/list/sprites = list()  // "foo_bar" -> list("32x32", 5)
+	var/list/cached_spritesheets_needed
+	var/generating_cache = FALSE
+	var/fully_generated = FALSE
+	/// If this asset should be fully loaded on new
+	/// Defaults to false so we can process this stuff nicely
+	var/load_immediately = FALSE
+	VAR_PRIVATE
+		// Kept in state so that the result is the same, even when the files are created, for this run
+		should_refresh = null
```

Comment:
```suggestion
	/// Kept in state so that the result is the same, even when the files are created, for this run
	VAR_PRIVATE/should_refresh = null
```
We avoid using relative pathing here.

Reply:
it makes me sad that you assume this is ported code :( we do the same thing!! and this was already here